tests/js/dictionary-tests.js

```diff
@@ -262,40 +267,15 @@ module.exports = {
             realm.write(() => { fields.field1=i } )
         }
 
-        return new Promise((resolve, reject) => {
-            setTimeout(() =>{
-                fields.removeAllListeners();
-                realm.close();
-                resolve()
-            }, 10000)
-        })
-    },
-
-    testDictionaryUserShouldNotDeleteFields() {
-        const DictSchema = {
-            name: "Dictionary",
-            properties: {
-                fields: "{}"
-            }
-        }
-
-        let realm = new Realm({schema: [DictSchema]})
-        realm.write(() => realm.create(DictSchema.name, {fields: {x1: 0, x2: 2}}))
-        let ff = realm.objects(DictSchema.name)[0]
-
-        delete ff.fields.x1
-        delete ff.fields.x2
-
-        TestCase.assertEqual(Object.keys(ff.fields)[0], "x2", "Should contain x2 field")
-        TestCase.assertEqual(Object.keys(ff.fields)[1], "x1", "Should contain x1 field")
-
-
         return new Promise((resolve, reject) => {
             setTimeout(() => {
+                TestCase.assertEqual(realm.objects(DictSchema.name)[0].fields.field1, 5);
+                TestCase.assertEqual(cnt,UPDATES+1,`We expect ${UPDATES+1} updates.`);
+                fields.removeAllListeners();
                 realm.close();
                 resolve();
-            }, 1000);
-        });
+            }, 2500)
```

Comment:
There must be a better way to do this than sleeping for 2.5seconds...

ditto elsewhere